worker/src/handlers/importPersonas/importPersonaHandler.js

```diff
@@ -98,28 +101,33 @@ export default personaService => async ({
   organisation
 }, done) => {
   // establish lock
-  const lock = await establishLock({ structure, data, organisation });
-  if (!lock) {
-    await promisify(publish)({
-      queueName: PERSONA_IMPORT_QUEUE,
-      payload: {
-        index,
-        data,
-        personaImportId,
-        structure,
-        organisation
-      }
-    });
-    done();
-    return;
-  }
+  let lock;
+  const afterValidationCb = async () => {
+    lock = await establishLock({ structure, data, organisation });
+    if (!lock) {
+      await promisify(publish)({
+        queueName: PERSONA_IMPORT_QUEUE,
+        payload: {
+          index,
+          data,
+          personaImportId,
+          structure,
+          organisation
+        }
+      });
+      done();
```

Comment:
If the call to `importPersona` does not throw an error below, this `done` callback will be called again on line 133 in this same file.